src/react/components/_finances/FinancialResources/HasAccessTo.js

```diff
@@ -0,0 +1,303 @@
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { activeRoles } from '../../../helpers/roles';
+import getLink from './getLink';
+
+const propTypes = {
+  dispatch: PropTypes.func.isRequired,
+  linkName: PropTypes.string,
+  links: PropTypes.object,
+  myAcademics: PropTypes.object,
+  myStatus: PropTypes.object,
+};
+
+const hasAccessToLink = (key, roles, careers, programs, delegate, summer) => {
+  const linkAccess = {
```

Comment:
Pull the `linkAccess` data outside the body of the function to prevent reevaluation of the literal every time the function is called.